packages/fastify/README.md

```diff
@@ -49,24 +53,92 @@ npm run build
 
 ## Usage
 
+Retrieve your Backend API key from the [API Keys](https://dashboard.clerk.dev/last-active?path=api-keys) screen in your Clerk dashboard and set it as an environment variable in a `.env` file:
+
+```sh
+CLERK_PUBLISHABLE_KEY=pk_*******
+CLERK_SECRET_KEY=sk_******
+```
+
+You will then be able to access all the available methods.
+
 ```javascript
-import { ClerkProvider } from '@clerk/nextjs';
-import { frFR } from '@clerk/fastify';
-
-function MyApp({ Component, pageProps }) {
-  return (
-    <ClerkProvider
-      localization={frFR}
-      {...pageProps}
-    >
-      <Component {...pageProps} />
-    </ClerkProvider>
-  );
-}
-
-export default MyApp;
+import 'dotenv/config'; // To read CLERK_API_KEY
+import Fastify from 'fastify';
+import { clerkPlugin, getAuth } from '@clerk/fastify';
+import type { FastifyInstance } from 'fastify';
+
+const server: FastifyInstance = Fastify({ logger: true });
+
+server.register(clerkPlugin);
+
+server.get('/private', async (req, reply) => {
+  const auth = getAuth(req);
+  if (!auth.userId) {
+    return reply.code(403).send();
+  }
+  return { hello: 'world' };
+});
+
+const start = async () => {
+  try {
+    await server.listen({ port: 3000 });
+  } catch (err) {
+    server.log.error(err);
+    process.exit(1);
+  }
+};
+start();
 ```
 
+### Advanced usage with scoped routes
+
```

Comment:
🙃 Please add a short description of this use case. I'd also suggest renaming the title to `Scoped routes` as in a big Fastify API this is going to be the most common setup since some paths will be protected, but others won't.

Reply:
Would this be okay?
> Support authenticated routes that the Clerk middleware will run as plugin in `preHandler` hook and unauthenticated routes that will not trigger the middleware using the Fastify [docs](https://www.fastify.io/docs/latest/Guides/Getting-Started/#loading-order-of-your-plugins).